flink-streaming-java/src/main/java/org/apache/flink/streaming/api/environment/StreamExecutionEnvironment.java

```diff
@@ -719,12 +719,24 @@ public void registerType(Class<?> type) {
 	 * time, event time, or ingestion time.
 	 *
 	 * <p>If you set the characteristic to IngestionTime of EventTime this will set a default
-	 * watermark update interval of 200 ms. If this is not applicable for your application
-	 * you should change it using {@link ExecutionConfig#setAutoWatermarkInterval(long)}.
+	 * watermark update interval of 200 ms. If this is not applicable for your application you
+	 * should change it using {@link ExecutionConfig#setAutoWatermarkInterval(long)}.
 	 *
 	 * @param characteristic The time characteristic.
+	 *
+	 * @deprecated In Flink 1.12 the default stream time characteristic has been changed to {@link
+	 *        TimeCharacteristic#EventTime}, thus you don't need to call this method for enabling
+	 * 		event-time support anymore. Explicitly using processing-time windows and timers works in
+	 * 		event-time mode. If you need to disable watermarks, please use {@link
+	 *        ExecutionConfig#setAutoWatermarkInterval(long)}. If you are using {@link
```

Comment:
I think it is useful to also mention explicitly in the doc the value to set the watermark interval in order to use `ProcessingTime`. Sth like: "If you need to disable watermarks, please set the {@code watermarkInterval} to 0 using the {@link ExecutionConfig#setAutoWatermarkInterval(long)}."

Reply:
Good point! I would update the Javadoc of `setAutoWatermarkInterval()` to mention that setting it to `0` will disable it. I'm reluctant to add it here because the message is already very long. WDYT?

Reply:
I forgot to say that the 
```
else {
			getConfig().setAutoWatermarkInterval(200); // TODO: 30.09.20 this is not needed now as the default value is 200
		}
``` 
is not strictly needed anymore, as `200` is the new default value.